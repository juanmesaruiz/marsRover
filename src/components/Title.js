import React from 'react';
import styled from 'styled-components';

const Title = ({ className }) => (
  <h1 className={className}>Mars Rover Mission</h1>
);

const StyledTitle = styled(Title)`
  margin: 0 0 30px;
`;

export default StyledTitle;
