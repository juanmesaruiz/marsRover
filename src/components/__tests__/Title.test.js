import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import Title from '../Title';

describe('Title test suite', () => {
  const component = <Title />;

  const setup = () => render(component);

  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(component, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Should match snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
