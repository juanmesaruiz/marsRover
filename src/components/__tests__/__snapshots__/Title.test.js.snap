// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title test suite Should match snapshot 1`] = `
.c0 {
  margin: 0 0 30px;
}

<div>
  <h1
    class="c0"
  >
    Mars Rover Mission
  </h1>
</div>
`;
